vowpalwabbit/cb_to_cb_adf.cc

```diff
@@ -0,0 +1,186 @@
+// Copyright (c) by respective owners including Yahoo!, Microsoft, and
+// individual contributors. All rights reserved. Released under a BSD (revised)
+// license as described in the file LICENSE.
+
+#include "cb_to_cb_adf.h"
+
+#include "vw_versions.h"
+#include "reductions.h"
+#include "learner.h"
+#include "vw.h"
+#include "cbify.h"
+#include "cb_algs.h"
+#include "cb_explore.h"
+#include "cb_label_parser.h"
+
+using namespace VW::LEARNER;
+using namespace VW::config;
+
+struct cb_to_cb_adf
+{
+  parameters* weights;
+  cbify_adf_data adf_data;
+  bool explore_mode;
+};
+
+template <bool is_learn>
+void predict_or_learn(cb_to_cb_adf& data, multi_learner& base, example& ec)
+{
+  data.adf_data.copy_example_to_adf(*data.weights, ec);
+
+  if (is_learn && !CB::is_test_label(&ec.l.cb))
+  {
+    uint32_t chosen_action = ec.l.cb.costs[0].action - 1;
```

Comment:
can `ec.l.cb.costs[0].action` here ever be `0` causing an overflow in `chosen_action`?

Reply:
I believe so, this is user input so it could be anything. chosen_action is an unsigned int and then right after on the next line we compare it against num_actions. If it overflows this check should fail.